src/Controls/src/Core/Handlers/NavigationPage/NavigationPageHandler.Android.cs

```diff
@@ -98,9 +102,10 @@ protected override void ConnectHandler(AView nativeView)
 			}
 
 			graph.StartDestination = destinations[0];
+
 			NavHost.NavController.SetGraph(graph, null);
 
-			for (var i = 1; i < destinations.Count; i++)
+			for (var i = NativeNavigationStackCount; i < NavigationStackCount; i++)
```

Comment:
When you set the graph the backstack automatically recreates so we're only checking here if we need to push more pages